README.md

```diff
@@ -1,101 +1,84 @@
 
-# How to use this repository?
-## Git LFS installation
-Download and install Git LFS using the instructions from [the Git LFS website](https://git-lfs.github.com)
+<!--We should motivate this better...-->
+Large distributed teams are able to efficiently work on shared codebases due to version control systems like Git. In an effort to [build ML models like we build open-source software](https://colinraffel.com/blog/a-call-to-build-models-like-we-build-open-source-software.html), Git-Theta is a Git extension that allows you to *efficiently* and *meaningfully* track a model's version history natively through Git.
 
-## Installing the git-theta package
-clone the repository
+# Example Usage
+Say you have a codebase for training an ML model along with a model checkpoint. Both the model and code are iteratively updated as new data is collected and new training methods are developed:
 ```bash
-git clone https://github.com/r-three/git-theta.git
+my_codebase
+├── model.pt
+└── train.py
 ```
-Install the git-theta package by running:
+Git-Theta allows you to use Git to track the version history of your code ***and*** model as you iteratively update them.
+
+In order to track the model checkpoint with Git-Theta, run the command
 ```bash
-cd git-theta
-pip install .[all]
+git theta track model.pt
 ```
 
-### A Single Deep Learning Framework
+This command signals to Git (by adding to `.gitattributes`) that when running certain operations on `model.pt`, it should delegate to Git-Theta.
 
-If you plan to track model checkpoints created by a single deep learning
-framework, for example only PyTorch or only Tensorflow, you can elect to only
-ensure the framework you use will be installed, avoiding the long install times
-and possible version requirements issues installing unused frameworks may bring.
+With the model tracked by Git-Theta, you can treat the model checkpoint exactly like you would any other file in a Git repository. All of the regular Git commands like `add`, `commit`, `push`, `pull`, `checkout`, `status`, `diff`, etc. will work on your model the way they would on any other file.
 
-For example, install git-theta with only pytorch checkpoint support:
+Additionally, when staging a change to a model, you can provide Git-Theta with additional information about ***what type*** of change is being staged (e.g., a sparse update, a low-rank update, etc.) by running `git theta add model.pt --update-type <update type>`. This allows Git-Theta to store the model update more efficiently, saving disk space and bandwidth when `push`-ing or `pull`-ing.
 
-``` bash
-cd git-theta
-pip install .[pytorch]
-```
+# Why is this better than using Git or Git LFS?
+Git on its own can certainly be used for versioning non-text files like model checkpoints. However, the main limiting factors are that
+1. Git remotes like Github and Bitbucket have a maximum file size (~50MB)
+2. Git is not designed to handle very large repositories
```

Comment:
A few more reasons.

3. A small change to a single model parameter shows as a change to the while model checkpoint.
4. Diffs on non-text files do not help localize which model parameters changed.

Also maybe this could be earlier as part of the extra motivation you mentioned.